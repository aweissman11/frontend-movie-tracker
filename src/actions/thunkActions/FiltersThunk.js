import * as Cleaners from '../../utilities/cleaners';
import fetchCall from '../../utilities/fetchCall';
import { setMovieList, setHasErrored, setIsOk, isLoading } from '../index';

export const getMovieList = (filters, query = '') => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const movies = await fetchCall(Cleaners.getFullUrl(filters, query));
      if (movies === 'failed') {
        dispatch(setIsOk(true));
      }
      dispatch(isLoading(false));
      dispatch(setMovieList(movies));
    } catch(e) {
      dispatch(setHasErrored(true));
    }
  }
}

