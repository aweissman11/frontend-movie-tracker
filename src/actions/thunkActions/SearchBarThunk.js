import * as Cleaners from '../../utilities/cleaners';
import fetchCall from '../../utilities/fetchCall';

export const getMovieList = (filterProperties, searchQuery) => {
  return async (dispatch) => {
    try {
      const movies = await fetchCall(Cleaners.getFullUrl(filterProperties, searchQuery));
      dispatch({
        type: 'GET_MOVIE_LIST',
        movies: movies
      })
    } catch(e) {
      dispatch({ type: 'ERROR'})
    }
  }
}

