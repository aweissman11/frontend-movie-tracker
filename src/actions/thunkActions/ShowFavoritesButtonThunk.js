import * as Cleaners from '../../utilities/cleaners';
import { setMovieList, setHasErrored, setIsOk, isLoading } from '../index';

export const getMovieList = (IDs) => {
  return async (dispatch) => {
    dispatch(isLoading(true))
    try {
      const results = await Cleaners.getFavoritesInfo(IDs);
      if (results === 'failed') {
        dispatch(setIsOk(true))
      }
      dispatch(isLoading(false))
      const movies = {
        results
      }
      await dispatch(setMovieList(movies))
    } catch(e) {
      dispatch(setHasErrored(true))
    }
  }
}

