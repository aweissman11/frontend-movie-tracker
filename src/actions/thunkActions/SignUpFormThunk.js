import { createNewUser } from '../../utilities/userDatabaseFetch';
import { isLoading, getUserLoggedIn, setHasErrored } from '../index';

export const submitNewUser = (name, email, password) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const newUserResponse = await createNewUser({ name, email, password });
      if (newUserResponse.error) {
        dispatch(setHasErrored(true));
      }
      dispatch(isLoading(false));
      await dispatch(getUserLoggedIn(newUserResponse.id, name));
    } catch(error) {
      dispatch(setHasErrored(true));
    }
  }
}