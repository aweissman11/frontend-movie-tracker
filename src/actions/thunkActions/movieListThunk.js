import * as Cleaners from '../../utilities/cleaners';
import { isLoading, setMovieList, setFavorites, setHasErrored, setIsOk } from '../index';
import fetchCall from '../../utilities/fetchCall';

export const getMovieList = () => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const movies = await fetchCall(Cleaners.movieListUrl);
      if (movies === 'failed') {
        dispatch(setIsOk(true));
      }
      dispatch(isLoading(false));
      dispatch(setMovieList(movies));
    } catch(e) {
      dispatch(setHasErrored(true));
    }
  }
}

export const updateFavorites = (id) => {
  return async (dispatch) => {
    try {
      const favorites = await Cleaners.getCurrentFavorites(id);
      dispatch(setFavorites(favorites));
    } catch(e) {
      dispatch(setHasErrored(true));
    }
  }
}