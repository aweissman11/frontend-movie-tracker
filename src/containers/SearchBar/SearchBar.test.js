import React from 'react'
import { SearchBar, mapDispatchToProps } from './';
import { getMovieList } from '../../actions/thunkActions/SearchBarThunk';

import { shallow } from 'enzyme';

describe('SearchBar', () => {
  let wrapper;
  const mockPreventDefault = jest.fn();

  const mockEvent = {
    preventDefault: mockPreventDefault,
    target: {
      name: 'searchInput',
      value: 'die hard'
    }
  }

  const mockSetFetchedMovies = jest.fn();
  const mockDispatch = jest.fn();
  const mockSubscribe = jest.fn();
  const mockGetState = jest.fn();

  const mockStore = {
    subscribe: mockSubscribe,
    dispatch: mockDispatch,
    getState: mockGetState
  }

  beforeEach(() => {
    wrapper = shallow(<SearchBar store={mockStore} setFetchedMovies={mockSetFetchedMovies}/>)
  });

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const expected = {
      searchInput: ''
    }

    expect(wrapper.state()).toEqual(expected);
  });

  it('should call handleChange on text input', () => {
    const event = {target: {
      value: 'a', 
      name: 'searchInput'
    }};
    const mockHandleChange = jest.fn();
    wrapper.instance().handleChange = mockHandleChange;

    wrapper.find('.search-input').simulate('change', event);

    expect(mockHandleChange).toHaveBeenCalledWith(event);
  });

  it('should set state when handleChange is called', () => {
    const event = {target: {
      value: 'a', 
      name: 'searchInput'
    }};

    wrapper.instance().handleChange(event);

    expect(wrapper.state().searchInput).toEqual('a');
  });

  describe('handleSubmitSearch', () => {
    it('should call preventDefault', () => {
      wrapper.instance().handleSubmitSearch(mockEvent);
      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it('should call setFetchedMovies with the correct parameters', () => {
      const mockState = {
        searchInput: 'die hard'
      }

      wrapper.instance().setState({ searchInput: mockState})
      wrapper.instance().handleSubmitSearch(mockEvent);

      expect(mockSetFetchedMovies).toHaveBeenCalledWith(mockState)
    });
  });


  // it('should map the dispatch to props on removeFavorites', () => {
  //   const mockDispatch = jest.fn();
  //   const actionToDispatch = jest.fn()
  //   const mappedProps = mapDispatchToProps(mockDispatch);
    
  //   mappedProps.setFetchedMovies()

  // expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  // })
});