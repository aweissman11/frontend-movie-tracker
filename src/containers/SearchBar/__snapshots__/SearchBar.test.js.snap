// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match the snapShot 1`] = `undefined`;
